### DISCUSSION TITLE
Actividad 1 - GRUPO 4

### DISCUSSION THREAD

**Discussion by MatyRusso22** [frlp-utn-ingsoft/catedra#89] [Show and tell]
https://github.com/MatyRusso22/vetsoft

**Comment by ivanduva** [comment]
Buenas chicos,
La actividad tiene algunos problemas que les vamos a pedir que arreglen para aprobarla:

- El form de producto se rompe al guardar cuando escribo un caracter no número en precio. Deberían filtrar eso desde el template.
- Lo mismo en el form de medicamentos con la dosis.
- El branch de medicamentos no está pusheado, deberían subirlo.

Cada cosa debería solucionarla el integrante que realizó la tarea. Tienen tiempo de arreglarlo hasta la fecha de entrega de la próxima actividad de test que la vamos a informar el lunes.

También les dejamos algunos comentarios que deberían tener en cuenta en las próximas actividades:
- De manera generalizada, vimos varios commits en donde los mensajes contienen información sobre archivos o funciones. Eso es redundante porque el commit ya tiene esa información. El mensaje debería contener una síntesis de la funcionalidad del commit sin necesidad de mencionar lo anterior.
- Vimos que en al menos una ocasión solucionaron mal conflictos y lo arreglaron en los commits siguientes. Por ejemplo https://github.com/MatyRusso22/vetsoft/commit/b316673f88ed42e036ebc9756cd443248f8554d0. Tengan cuidado la próxima porque en el commit de merge la app queda rota. La idea de solucionar conflictos es hacerlo en el commit de merge, no en varios commits.
- Valoramos que hayan intentado estandarizar los branches, pero vemos que hay algunos que parecen duplicados porque tienen un nombre muy parecido. Si no son necesarios pueden borrarlos.
- En al menos una ocasión (modelo medicamento), hicieron un commit con la migración luego de haber desarrollado toda la funcionalidad. Entendemos que tal vez se hayan olvidado de subirlo hasta el final, pero el problema es que en los commits anteriores la app podría no funcionar. Tengan cuidado con esto la próxima.
- En varios casos hay muchos commits seguidos con el mismo nombre. Por ejemplo: https://github.com/MatyRusso22/vetsoft/commit/7bfac54ca4a02dd71b4703d06789f9b3f1a48665 o https://github.com/MatyRusso22/vetsoft/commit/8ee4938e23e498675602d3c0fb0fc1d55de7e22f. Si fue intencional, no es una buena práctica. También puede haber sido porque tuvieron que solucionar conflictos y copiaron el mismo mensaje que el commit anterior en el commit de merge. En ese caso les recomendamos que dejen el commit default de git. Si no se entiende o si tienen dudas pueden consultarnos.
- También vimos commits que agregan archivos vacíos, lo cual no tiene mucho sentido y es un poco confuso porque no agrega ninguna funcionalidad.

Saludos

**Reply by MatyRusso22** [reply]
Ahi se realizaron las correcciones profe, volvemos a entregar el tp 1: https://github.com/MatyRusso22/vetsoft

**Reply by ivanduva** [reply]
Buenas,
El branch de medicamento lo veo pusheado, así que eso está ok.
Pero los otros puntos no solo no se solucionaron, sino que agregaron más problemas:

- Ambos forms de producto y medicamento tienen `None` por default en el campo de precio y dosis respectivamente.
- Ambos campos permiten ingresar cualquier caracter
- El de producto se rompe al editar si pongo cualquier caracter en precio
- El de medicamento se rompe al crear y al editar si pongo cualquier caracter en precio

Cuando arreglan ujn bug, prueben que efectivamente esté arreglado. El problema no es tanto que existan bugs, sino que no corran la aplicación o no la prueben correctamente.
Tienen hasta el 26 para hacer los arreglos. Cualquier cosa si necesitan ayuda, nos pueden preguntar.
Saludos

**Comment by ruipablo** [comment]
Buenas profe, ya se realizaron las correcciones.
https://github.com/MatyRusso22/vetsoft

**Reply by ivanduva** [reply]
Ahora se rompe el form de medicamentos si pongo un float en dosis porque en el modelo el atributo es entero. Deberían permitir solo enteros en el form.

**Comment by ruipablo** [comment]
Bunas profe ya esta corregido dosis
https://github.com/MatyRusso22/vetsoft

**Reply by ivanduva** [reply]
Buenas. 
La entrega está fuera de término. Recuerden que como explicamos en las condiciones de aprobación, si entregan fuera de término 2 veces quedan afuera de la promoción. Por ahora tienen 1.
Van a tener que corregir de nuevo porque al editar está creando una instancia nueva de Medicine, por lo tanto no funciona la edición. Debería hacer un update de la instancia ya creada. Como dije antes, el problema no es que haya bugs sino que no prueben la app o que no noten que el ABM de medicina no está funcionando.

**La última fecha límite de recuperatorio, tanto para esta como las demás actividades, será el último día de cursada antes de las vacaciones: Lunes 8 de Julio 23:59 hs.**

Saludos

**Reply by MatyRusso22** [reply]
Buenas profe, ya solucionamos el bug. Saludos

**Comment by ivanduva** [comment]
Bien, queda aprobado.
Saludos
